apps/settings/js/modules/sim_security.js

```diff
@@ -0,0 +1,85 @@
+/**
+ * This is a simple model that can help us mimic the status change event
+ * when enabling / disabling simcard.
+ *
+ * @module SimSecurity
+ */
+define(function(require) {
+  'use strict';
+
+  var Module = require('modules/base/module');
+  var SettingsUtils = require('modules/settings_utils');
+  var EventEmitter = require('modules/base/event_emitter');
+
+  var APIS = [
+    'unlockCardLock',
+    'setCardLock',
+    'updateContact',
+    'getCardLock',
+    'getCardLockRetryCount'
+  ];
+
+  var EVENT_CHANGE_APIS = [
+    'unlockCardLock',
+    'setCardLock'
+  ];
+
+  var SimSecurity = Module.create(function() {
+    this.super(EventEmitter).call(this, [
+      'sim-security-pin-enabled',
+      'sim-security-pin-disabled',
+      'sim-security-pin2-enabled',
+      'sim-security-pin2-disabled',
+      'sim-security-fdn-enabled',
+      'sim-security-fdn-disabled',
+      'sim-security-pin-changed',
+      'sim-security-pin2-changed'
+    ]);
+  }).extend(EventEmitter);
+
+  APIS.forEach(function(apiName) {
+    SimSecurity.prototype[apiName] = function(cardIndex) {
+      if (arguments.length < 2) {
+        this.throw(
+          'You are using SimSecurity API wrongly and you may forget' +
+          'to put your cardIndex, please check again');
+      }
+
+      var passedArguments = [].slice.call(arguments, 1);
+      var icc = SettingsUtils.getIccByCardIndex(cardIndex);
+
+      if (!icc) {
+        console.error('We can\'t find needed icc object');
+        return Promise.reject();
+      } else {
+        var func = icc[apiName];
+        var promise = func.apply(icc, passedArguments);
+        return promise.then((result) => {
+          if (EVENT_CHANGE_APIS.indexOf(apiName) > -1) {
+            // we don't have to check passing arguments
+            // because we are under Promise.then()
+            var status = '';
+            var lockType = passedArguments[0].lockType;
+            var enabled = passedArguments[0].enabled;
+            var newPin = passedArguments[0].newPin;
+
+            // If we have newPin, it means that we are changing pin or pin2
+            if (newPin) {
+              status = 'changed';
+            } else {
+              status = enabled ? 'enabled' : 'disabled';
+            }
+            this._emitEvent('sim-security-' + lockType + '-' + status,
+              cardIndex);
+          }
+          return Promise.resolve(result);
+        }, (error) => {
+          return Promise.reject(error);
```

Comment:
This can be omitted if you only want to return a rejection. (it returns a rejection by default if the reject function is not provided)